### DISCUSSION TITLE
How to count the number of visits to the data in the collection?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#4713] [Feature Requests]
For example, How many times an article in the `article collection` has been read.

### The solution I know so far

a. Through `Hooks`, but there is a problem, the `date_updated` field is updated every time the count is updated.

b. By checking the database, I don't know if it is possible to get the number of queries of the collection by requesting directus_activity. I didn't try it. This is too troublesome.

c. Creating an `article_reads` collection for `article collection` (many to one), and each request creates a piece of data in the `article_reads` collection.This will generate a lot of useless data.

The b and c, Sometimes this method has a big disadvantage because of manual intervention.

### Desired result

I can add a count field, +1 each time the interface is requested (optional: multiple requests for a single IP are counted as only one)

Hope you can understand what I'm talking about.

**Comment by benhaynes** [comment]
The only way I can think to do this as of now is to store that count on a different Collection. There's no way to "skip" the `date_updated` process manually, unless you make the update to the database directly (as SQL).

**Reply by cdwmhcc** [reply]
Are there any plans? I think this feature is useful in more cases.😊😊😊😊

**Reply by benhaynes** [reply]
No, not beyond what I just mentioned.

**Reply by cdwmhcc** [reply]
Sorry, I don't understand what you mean,Do you suggest that I have used the database directly to update it?😊

**Reply by benhaynes** [reply]
Correct, or just store the metrics in a different table/collection so it doesn't trigger the auto-updated fields.

**Reply by cdwmhcc** [reply]
I seem to understand, I will study it, thank you boss for your help.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
